helix-core/src/main/java/org/apache/helix/customizedstate/CustomizedStateProvider.java

```diff
@@ -61,29 +61,16 @@ public synchronized void updateCustomizedState(String customizedStateName, Strin
    * Update a specific customized state based on the resource name and partition name. The
    * customized state is input as a map
    */
-  public synchronized void updateCustomizedState(String customizedStateName, String resourceName,
+  public void updateCustomizedState(String customizedStateName, String resourceName,
       String partitionName, Map<String, String> customizedStateMap) {
     PropertyKey.Builder keyBuilder = _helixDataAccessor.keyBuilder();
     PropertyKey propertyKey =
         keyBuilder.customizedState(_instanceName, customizedStateName, resourceName);
     ZNRecord record = new ZNRecord(resourceName);
-    Map<String, Map<String, String>> mapFields = new HashMap<>();
-    CustomizedState existingState = getCustomizedState(customizedStateName, resourceName);
-    if (existingState != null
-        && existingState.getRecord().getMapFields().containsKey(partitionName)) {
-      Map<String, String> existingMap = new HashMap<>();
```

Comment:
It is not clear what the original code tries to do here. It seems trying to merge the map. But not really doing so. Can you confirm the original logic is right or not?

The concern is that the default update method is doing deep merge. That is if you have an existing state map with some stale states, even after merging, those stale states will be kept in the map. But I assume we should clean those stale states for updating?

Reply:
So this is user to update a customized state for a certain partition. I don't think we have the knowledge to clean up stale states, meaning if the user only update "CURRENT_STATE" of customized state, we update it, and keep the other fields unchanged. If they would like to do a full update, they will use the other API and provide the full map, and the original map will be overwritten.

Reply:
I don't think you get my point, or maybe I did not get the point. Anyway, let's go with an example,
1. The original map {"CSTypeA":"foo", "CSTypeB":"bar"}
2. User try to update with map {"CSTypeC":"foo", "CSTypeD":"bar"}
3. I assume the expect result for the custmize state is {"CSTypeC":"foo", "CSTypeD":"bar"}.
4. But really by the default merge method we get: {"CSTypeA":"foo", "CSTypeB":"bar", "CSTypeC":"foo", "CSTypeD":"bar"}
CSTypeA and CSTypeB is the stale state that I meant.

Reply:
Yeah, we do want the full map like instead of replacing the previous map. Because for customized state, the key are fixed, e.g. "previous state", "current state", etc. We do not allow user to delete entries, and if they do not have value, the entry will have null value. 

For this API, it's updating the per partition customized state, not adding new type of customized states.
So it's a definite map of
{"start time": xxx
"end time:xxx
"previous state":xxx
"current state":xxx
}
and customers update one entry or the full map.

Reply:
I see. That makes sense.